FWCore/Services/plugins/SimpleMemoryCheck.cc

```diff
@@ -439,16 +551,22 @@ namespace edm {
       descriptions.add("SimpleMemoryCheck", desc);
     }
 
-    void SimpleMemoryCheck::openFiles() {
+    std::optional<std::string> SimpleMemoryCheck::openFilesNoThrow() {
 #ifdef LINUX
-      if (monitorPssAndPrivate_) {
-        std::ostringstream smapsNameOst;
-        smapsNameOst << "/proc/" << getpid() << "/smaps";
-        if ((smapsFile_ = fopen(smapsNameOst.str().c_str(), "r")) == nullptr) {
-          throw Exception(errors::Configuration) << "Failed to open smaps file " << smapsNameOst.str() << std::endl;
-        }
+      std::ostringstream smapsNameOst;
+      smapsNameOst << "/proc/" << getpid() << "/smaps";
+      if ((smapsFile_ = fopen(smapsNameOst.str().c_str(), "r")) == nullptr) {
+        return smapsNameOst.str();
       }
 #endif
+      return {};
+    }
+
+    void SimpleMemoryCheck::openFiles() {
+      auto smapsFile = openFilesNoThrow();
+      if (smapsFile.has_value()) {
```

Comment:
Shouldn't this be
```suggestion
      if (not smapsFile.has_value()) {
```

Reply:
Perhaps confusingly I made the `openFilesNoThrow()` to return a value of the file name if the open failed. The value is only used here in the exception message.

Would adding a comment to `openFilesNoThrow()` and renaming the `smapsFile` to e.g. `smapsFileIfOpenFailed` be clear enough?

Reply:
Improving the documentation is sufficient.